src/components/nuxt-icon.vue

```diff
@@ -0,0 +1,76 @@
+<template>
+  <IconComponent
+    :class="{
+      'nuxt-icon': fontControlled,
+      'nuxt-icon--fill': !filled
+    }"
+  />
+</template>
+
+<script lang="ts">
+import { h } from '#imports'
+
+const iconsImport = import.meta.glob('assets/icons/**/**.svg', {
+  import: 'default',
+  eager: true,
+  query: {
+    component: ''
+  }
+})
+
+function IconNotFound(name: string) {
+  return {
+    setup() {
+      return () =>
+        h('span', { innerHTML: `<!-- icon with name "${name}" not found -->` })
+    }
+  }
+}
+</script>
+
+<script setup lang="ts">
+// file imported from https://github.com/gitFoxCode/nuxt-icons/blob/89e53649e5868c31fc97869918ede96504ae1a04/src/runtime/components/nuxt-icon.vue
+// with some modifications
+import { markRaw, watch } from '#imports'
+
+const props = withDefaults(
+  defineProps<{
+    name: string
+    fontControlled?: boolean
+    filled?: boolean
+  }>(),
+  { filled: false, fontControlled: true }
+)
+
+const IconComponent = markRaw(
+  iconsImport[`/assets/icons/${props.name}.svg`] || IconNotFound(props.name)
+)
+
+watch(
+  () => props.name,
+  () => {
+    const component = iconsImport[`/assets/icons/${props.name}.svg`]
+    if (!component) {
+      console.error(
+        `[nuxt-svgo] Icon '${props.name}' doesn't exist in 'assets/icons'`
+      )
+      IconComponent.value = IconNotFound(props.name)
+    } else {
+      IconComponent.value = component
+    }
+  }
+)
+</script>
+
+<style>
```

Comment:
I agree with scoping, I have applied it in a commit
but since we are using the same component name and folder structure (assets/icons), I don't think they will be able to use this component effectively together with nuxt-icons
besides, nuxt-icons module is not really a good way for using svgs anyway :)

Reply:
weird, scoping it somehow breaks the tests, checking it now

Reply:
I see why its breaking tests, because it's adding a `data-v-[random_str]` attribute to the rendered component

@cpsoinos any ideas how to test it with this?

Reply:
done. removed the openning tag from expect of tests